features/FxA-83-sign-in-confirmation-email/README.md

```diff
@@ -0,0 +1,169 @@
+for Sync sign-ins,
+fewer accounts will be compromised.
+We will know this to be true if,
+in the event of an attack,
+the number of successful sign-ins remains constant
+and the count of 104 (unverified account) errors
+from protected endpoints
+increases in line with any wider increase in traffic.
+
+## Assumptions
+
+* The `/account/login`
+  and `/password/change/start` endpoints
+  are responsible for initiating
+  the confirmation process.
+
+* The following endpoints should be protected:
+  * `/certificate/sign`
+  * `/account/keys`
+  * `/account/delete`
+
+* In most cases,
+  attackers do not have access
+  to our users' email accounts.
+  Users that do have a compromised email account
+  are out of scope for this feature.
+
+## Constraints
+
+* In light of the recent attack,
+  the solution should be quick to implement.
+
+* The solution must work across all clients,
+  without patches landing in client code.
+
+* The solution must not affect existing sessions
+  from the user's point of view.
+
+* The solution must work from
+  regular Sync sign-in,
+  `/force_auth` and
+  when signing-in to Sync from `/signup`.
+
+## Proposed solution
+
+Add a notion of “verified”
+to sessionTokens and keyFetchTokens.
+Tokens that are unverified have reduced powers,
+meaning that requests to protected endpoints will fail
+with error number 102 (unverified user).
+Token verification is achieved
+by following a link sent by email.
+
+## User stories
+
+* As a Sync user,
+  when signing in to my Firefox Account,
+  I want my identity confirmed
+  via my email account.
+
+* As a user of an OAuth relier,
+  when signing in to my Firefox Account,
+  I do not want an extra confirmation step.
+
+## Work breakdown
+
+### fxa-content-server
+
+- [ ] Add strings to `strings.js`
+  to get localization done quicker.
+- [ ] Implement "confirm your email" screen.
+- [ ] Implement confirmation landing screen.
+- [ ] Add handling for `verified` and `challenge` fields
+  in `/account/login`
+  and `/password/change/start` responses.
+- [ ] Add handling for 102 errors
+  in `/certificate/sign`,
+  `/account/keys`
+  and `/account/delete`.
+
+### fxa-auth-server
+
+- [ ] Add method to `mailer.js`
+  for sending confirmation email.
+- [ ] Modify `/account/login`
+  and `/password/change/start`
+  to send verification status to db,
+  initiate verification email
+  and set `challenge` on response.
+- [ ] Create a `verifiedSessionToken`
+  auth strategy for Hapi
+  that fails with 102 (unverified user) error
+  if sessionToken is not verified.
+- [ ] Create a `verifiedKeyFetchToken`
+  auth strategy for Hapi
+  that fails with 102 (unverified user) error
+  if keyFetchToken is not verified.
+- [ ] In `/account/delete`,
+  use new `verifiedSessionToken` auth strategy.
+- [ ] In `/certificate/sign`,
+  use new `verifiedSessionToken` auth strategy.
+- [ ] In `/account/keys`,
+  use new `verifiedKeyFetchToken` auth strategy.
+- [ ] Add a new endpoint for token verification.
```

Comment:
You'll also need to ensure the tokens generated by `/account/create` get automagically marked as verified when the user verifies their account.